tools/packaging/static-build/ovmf/build-ovmf.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 IBM
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+source "${script_dir}/../../scripts/lib.sh"
+
+ovmf_repo="${ovmf_repo:-}"
+ovmf_dir="edk2"
+ovmf_version="${ovmf_version:-}"
+kata_version="${kata_version:-}"
+
+set +u
+ovmf_build="${ovmf_build:-sev}"
+architecture="X64"
+toolchain="GCC5"
+build_target="RELEASE"
+
+if [ -z "$ovmf_repo" ]; then
+       info "Get ovmf information from runtime versions.yaml"
+       ovmf_url=$(get_from_kata_deps "externals.ovmf.url" "${kata_version}")
+       [ -n "$ovmf_url" ] || die "failed to get ovmf url"
+       ovmf_repo="${ovmf_url}.git"
+fi
+
+[ -n "$ovmf_repo" ] || die "failed to get ovmf repo"
+
+if [ "${ovmf_build}" == "sev" ]; then
+       [ -n "$ovmf_version" ] || ovmf_version=$(get_from_kata_deps "externals.ovmf.sev.version" "${kata_version}")
+fi
+[ -n "$ovmf_version" ] || die "failed to get ovmf version or commit"
+
+info "Build ${ovmf_repo} version: ${ovmf_version}"
+
+[ -d "${ovmf_dir}" ] || git clone "${ovmf_repo}"
+cd "${ovmf_dir}"
+git checkout "${ovmf_version}"
+git submodule init
+git submodule update
+
+info "Using BaseTools make target"
+make -C BaseTools/
+set -u
+
+info "Calling edksetup scipt"
+# disabling set -u because edksetup.sh attempts to expands undefined variables
+set +u
+source edksetup.sh
+
+if [ "${ovmf_build}" == "sev" ]; then
+       info "Creating dummy grub file"
+       #required for building AmdSev package without grub
+       touch OvmfPkg/AmdSev/Grub/grub.efi
+fi
+
+
+#todo: check that this is even valid shell
+if [ "${ovmf_build}" == "sev" ]; then
+       [ -n "$ovmf_package" ] || ovmf_package=$(get_from_kata_deps "externals.ovmf.sev.package" "${kata_version}")
```

Comment:
Move this (and the other initializations below) up at the begin of the script so that if some information for the build is missing then it exists right away (fail fast).